### PR TITLE
Implemented eth_getTransactionByHash

### PR DESCRIPTION
**Pull request by ntallar**
## Description

Includes the implementation of the `eth_getTransactionByHash` JSON RPC method.
In order to do so, a `TransactionMappingStorage` was created which stores the mapping `TxHash -> (BlockHash, TxIndex)`.

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/db/storage/TransactionMappingStorageSuite.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      val txMappingStorage = txHashes.zip(txLocationList).foldLeft(initialTxMappingStorage){
+        case (recTxMappingStorage, (txHash, txLocation)) =>
+          recTxMappingStorage.put(txHash, txLocation)
+      }
+
+      txHashes.zip(txLocationList).foreach{case (txHash, txLocation) => assert(txMappingStorage.get(txHash).contains(txLocation)) }
+    }
+  }
+
+  test("TotalDifficultyStorage delete") {
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/db/storage/TransactionMappingStorageSuite.scala]: Are those test descriptions accurate? I don't see anything about difficulty in them. Same goes for the comments.

File: src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala [hunk 2]
```diff
@@ -672,6 +673,40 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
...
+    val request = GetTransactionByHashRequest(txToRequestHash)
+    val response = ethService.getTransactionByHash(request)
+
+    pendingTransactionsManager.expectMsg(PendingTransactionsManager.GetPendingTransactions)
+    pendingTransactionsManager.reply(PendingTransactions(Nil))
+
+    response.futureValue shouldEqual Right(GetTransactionByHashResponse(None))
+  }
+
+  it should "handle get transaction by hash if the tx is still pending" in new TestSetup {
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: 👍

File: src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala [hunk 3]
```diff
@@ -672,6 +673,40 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
     response.futureValue shouldEqual Right(GetTransactionCountResponse(BigInt(999)))
   }
 
+  it should "handle get transaction by hash if the tx is not on the blockchain" in new TestSetup {
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: "... not on the blockchain and not in the tx pool", right?

File: src/main/scala/io/iohk/ethereum/db/storage/TransactionMappingStorage.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import akka.util.ByteString
+import io.iohk.ethereum.db.dataSource.DataSource
+import io.iohk.ethereum.db.storage.TransactionMappingStorage.{TransactionLocation, TxHash}
+import boopickle.Default._
+
+class TransactionMappingStorage(val dataSource: DataSource) extends KeyValueStorage[TxHash, TransactionLocation, TransactionMappingStorage] {
+
+  val namespace: IndexedSeq[Byte] = Namespaces.TransactionMappingNamespace
+  def keySerializer: TxHash => IndexedSeq[Byte] = identity
+  def valueSerializer: TransactionLocation => IndexedSeq[Byte] = Pickle.intoBytes(_).array()
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/db/storage/TransactionMappingStorage.scala]: Just curious, why not use our good friend RLP? :)

Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/db/storage/TransactionMappingStorage.scala]: That was my first idea but as there is a task for removing the use of it for storages (https://iohk.myjetbrains.com/youtrack/issue/EC-119), `boopickle` serializing is also used for `FastSyncStateStorage` and it required the same effort decided to use the `boopickle` serializing.

**Pull request merged**